### ISSUE TITLE
Special linting for TSV tN Files

### ISSUE THREAD

**Issue by jag3773** [unfoldingWord-dev/door43.org#873]
## Story

As a *tC user* I want to see the correctly highlighted text so that I'm not confused when checking my translation.

## Notes

The tN TSV files are [structured](https://git.door43.org/unfoldingWord/en_tn/#structure) in such a way that they are intended to quote the UGNT text in the `OrigQuote` field.  During the linting process we should verify that all of the OrigQuote entries in the TSV file(s) match what is in the [published version of the UGNT](https://git.door43.org/Door43-Catalog/el-x-koine_ugnt).

I think the tC team has some code that may be useful to use here.

cc @benjore @mannycolon @richmahn @RoyalSix @PhotoNomad0 @RobH123

**Comment by RobH123** [comment]
@jag3773 The tX converter part of the system that does the linting has been made to be independent of knowledge about Door43. You pass it a set of files to convert and it passes back the results. Even the manifest file for tN contains no link pointing to the version of UGNT that was used. So it either has to be hard-coded or passed in to the system somehow.

Alternatively, it could be done in the preprocessor in the Door43 code but that's not ideal either.

**Comment by RobH123** [comment]
A nice thing would be to run this linter not only when tN has an update pushed, but also when UGNT has an update pushed. (Don't see an easy way to implement this though.)

**Comment by jag3773** [comment]
> Even the manifest file for tN contains no link pointing to the version of UGNT that was used. So it either has to be hard-coded or passed in to the system somehow.

I think that we should make a constraint by convention to require that the Greek and Hebrew referenced in the OrigQuote field must be listed in the `relation` field of the manifest.  Would that work?  It would still be a list that would have to be iterated through but it seems like that could solve the problem here?

**Comment by RobH123** [comment]
Yes, that would be the best way ahead I think. And for other reasons, I think that `relation` field should include a date or version number even though that requires more manual maintenance. (Also for other reasons, but maybe helpful to my comment just above, maybe the original texts could contain something like a `quoted_in` field.)

**Comment by RobH123** [comment]
Ok, I done a first pass on this that processes both OT and NT quotes. See https://dev.door43.org/u/unfoldingWord/en_tn/master/ (which has Ruth and Ezra suppressed so more NT errors are shown).
